### PR TITLE
Fixed broken links to MSDN COM interop topics

### LINKED ISSUE #3071: Links on page don't seem to be working - Marshaling data with COM interop
# Issue Title
Links on page don't work - Link: https://github.com/dotnet/docs/blob/master/docs/framework/interop/marshaling-data-with-com-interop.md

## Issues with Existing Topics

The links in the page point back to msdn articles which don't exist.

### PR DESCRIPTION
**Pull request by rpetrusha**
# Fixed broken links to MSDN COM interop topics  

## Summary

Most of the links in this topic were to topics not in the .NET Framework 4.5 doc set, since they were "paved over" and not migrated to CAPS. I've replaced them with links to the archived content in the .NET Framework 4 doc set on MSDN.

Fixes #3071 

## Suggested Reviewers

@mairaw @pieandcakes

### CODE REVIEW

File: docs/framework/interop/marshaling-data-with-com-interop.md [hunk 1]
```diff
@@ -32,30 +30,30 @@ COM interop provides support for both using COM objects from managed code and ex
   
 -   [Type Library Exporter (Tlbexp.exe)](../../../docs/framework/tools/tlbexp-exe-type-library-exporter.md), which produces a COM type library from an assembly and generates a wrapper that performs marshaling during method calls.  
   
- This section describes the processes for customizing interop wrappers when you can (or must) supply the marshaler with additional type information.  
+ The following topics describe the processes for customizing interop wrappers when you can (or must) supply the marshaler with additional type information.  
   
 ## In This Section  
- [COM Data Types](http://msdn.microsoft.com/en-us/f93ae35d-a416-4218-8700-c8218cc90061)  
+ [COM Data Types](https://msdn.microsoft.com/en-us/library/sak564ww(v=vs.100).aspx)
```
Comment (mairaw) [inline docs/framework/interop/marshaling-data-with-com-interop.md]: as I just commented on the issue, these links are not part of this topic TOC any longer, I think this section should be updated to actually reflect what's on the TOC and useful links should be moved to the next section.

**Review by mairaw — CHANGES_REQUESTED** [review]
Updated links look good but I've left a comment re. which links should be under in this section.

**Comment by pieandcakes** [comment]
LGTM

File: docs/framework/interop/marshaling-data-with-com-interop.md (hunk 1 above)
Comment (rpetrusha) [reply docs/framework/interop/marshaling-data-with-com-interop.md]: @mairaw, I've updated the two sections.

File: docs/framework/interop/marshaling-data-with-com-interop.md [hunk 2]
```diff
@@ -32,30 +30,33 @@ COM interop provides support for both using COM objects from managed code and ex
...
   
- This section describes the processes for customizing interop wrappers when you can (or must) supply the marshaler with additional type information.  
+ The following sections link to topics that describe the processes for customizing interop wrappers when you can (or must) supply the marshaler with additional type information.  
   
 ## In This Section  
- [COM Data Types](http://msdn.microsoft.com/en-us/f93ae35d-a416-4218-8700-c8218cc90061)  
+[How to: Create Wrappers Manually](how-to-create-wrappers-manually.md)   
+Describes how to create a COM wrapper manually in managed source code. 
+ 
+ [How to: Migrate Managed-Code DCOM to WCF](../../../docs/framework/interop/how-to-migrate-managed-code-dcom-to-wcf.md)
```
Comment (mairaw) [inline docs/framework/interop/marshaling-data-with-com-interop.md]: nit: should this be listed first to match TOC order?

**Review by mairaw — APPROVED** [review]
LGTM apart from a small nit

File: docs/framework/interop/marshaling-data-with-com-interop.md (hunk 2 above)
Comment (rpetrusha) [reply docs/framework/interop/marshaling-data-with-com-interop.md]: @mairaw, yes, the orders should match. I've changed the order in the TOC.

**Pull request merged**